destination_middleware.go

```diff
@@ -187,6 +188,11 @@ func (d *destinationWithBatch) Configure(ctx context.Context, config config.Conf
 		return fmt.Errorf("invalid %q: must not be negative", configDestinationBatchDelay)
 	}
 
+	// TODO: emit this warning once we move to source batching
```

Comment:
Since source connector can now emit batches, we should think about deprecating the middleware on the destination, as batching on the source results in a much better performance.

That said, we shouldn't do it just yet, since Conduit currently still handles records one by one. Only once we adopt the new internal architecture, which actually knows how to handle batches, should we uncomment this part and deprecate the middleware.